lib/reline/ansi.rb

```diff
@@ -235,7 +235,7 @@ def self.get_screen_size
     s = [ENV["LINES"].to_i, ENV["COLUMNS"].to_i]
     return s if s[0] > 0 && s[1] > 0
     [24, 80]
-  rescue Errno::ENOTTY
+  rescue Errno::ENOTTY, Errno::ENODEV, Errno::EBADF
```

Comment:
I think adding `Errno::ENODEV` is enough.

The main problem of #690 is that just requiring 'reline' was calling `IO#winsize`. #703 will fix it.

We need to rescue `Errno::ENOTTY` and `Errno::ENODEV` to make reline work in these two scenario.
```
# For Errno::ENODEV
ruby -rreline -e "Reline.readline" < /dev/null
```

```
# For Errno::ENOTTY
ruby -rreline -e "Reline.readline" < file
```

But for `Errno::EBADF`, I think we don't need to rescue it. Just like we don't rescue IOError here.
```ruby
STDIN.close
STDIN.winsize #=> closed stream (IOError)
Reline.readline #=> `tty?': closed stream (IOError)
```

Reply:
Ok, just catching `Errno::ENOTTY` and `Errno::ENODEV`